asset/asset.go

```diff
@@ -245,6 +245,11 @@ func (i ID) String() string {
 	return hex.EncodeToString(i[:])
 }
 
+// IsEqual returns true if the ID matches the provided ID.
+func (i ID) IsEqual(a ID) bool {
```

Comment:
just a drive-by comment: `asset.ID` is an array, so it supports the `==` operator.